### PR TITLE
Fix and simplify HandConstraint solver tracking

### PR DESCRIPTION
**Pull request by Troy-Ferrell**
## Overview
HandConstraint is simplified to follow properties set in SolverHandler instead of overriding and tracking hands itself. Further, the UnityEvents have been updated to fire properly.

- HandConstraint now leverages SolverHandler's ability to track/switch between particular hands. 
- HandConstraint can choose to ignore the tracked hand if it considers it not valid or request SolverHandler to switch hands (if possible) via PreferredTrackedHandedness
- Added PreferredTrackedHandedness property which indicates which hand to default to when both are available
- UnityEvents now allocated by default so easier extension via code
- OnHandActivate/Deactivate fire whenever a hand is supported or not
- Created HandednessExtensions to simplify code interaction
- HandMenu prefab updated to hide onFirstHandFound and onLastHandLost

Related PR: #6311 

## Changes
- Fixes: #6430

### CODE REVIEW

File: Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/SolverHandler.cs [hunk 1]
```diff
@@ -453,18 +481,21 @@ private Transform GetControllerRay(Handedness handedness)
...
+            if (trackingTarget == null)
+            {
+                return true;
+            }
 
             // If we were tracking a particular hand, check that our transform is still valid
             // The HandJointService does not destroy it's own hand joint tracked GameObjects even when a hand is no longer tracked
             // Those HandJointService's GameObjects though are the parents of our tracked transform and thus will not be null/destroyed
-            if (this.TrackedTargetType == TrackedObjectType.HandJoint && this.currentTrackedHandedness != Handedness.None)
+            if (TrackedTargetType == TrackedObjectType.HandJoint && currentTrackedHandedness != Handedness.None)
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/SolverHandler.cs]:
```suggestion
            if (TrackedTargetType == TrackedObjectType.HandJoint && !currentTrackedHandedness.IsNone())
```

**Comment by Troy-Ferrell** [comment]
@gilbdev FYI

**Comment by Troy-Ferrell** [comment]
> Thank you so much for updating the HandConstraint, the HandednessExtensions is gold too! One bug I did notice when running HandMenuExamples scene is the menu would not alternate between hands if the below steps where followed:
> 
> 1. Run the HandMenuExamples scene
> 2. Bring the left hand into view and rotate it so the palm faces the camera (the menu appears)
> 3. Bring the right hand into view and rotate it so the palm faces the camera
> 4. Rotate the left hand so the palm faces away from the camera
> 
> Result: the hand menu disappears from the left hand but does not re-appear on the right hand even though it is facing the camera.
> 
> The expected result is that the menu would move to the right hand and continue to constrain to "active" hands.

Fixed!

**Pull request merged**